### PR TITLE
vtk_8: remove

### PR DESCRIPTION
**Pull request by bcdarwin**
## Description of changes

Somewhat aggressive removal of vtk_8 and depending packages; basically:
 - bump `mirtk` to an unstable version;
 - remove `opencascade` since development seems to have switched back to `opencascade-occt` (e.g. pythonocc-core and cadquery switched over; see also https://github.com/tpaviot/oce/issues/745#issuecomment-992285943) and handle some resulting breakage.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -24461,9 +24457,6 @@ with pkgs;
     python = python3;
   };
 
-  opencascade = callPackage ../development/libraries/opencascade {
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix:24464]: Will break eval without aliases https://github.com/NixOS/nixpkgs/blob/1774d07242995050d2d8a91cb4da0855eac2e472/pkgs/applications/science/physics/elmerfem/default.nix#L1

Comment (bcdarwin) [reply pkgs/top-level/all-packages.nix:24464]: Fixed evaluation.

Comment (bcdarwin) [reply pkgs/top-level/all-packages.nix:24464]: Sorry.

**Comment by bcdarwin** [comment]
Results of `nixpkgs-review` on x86_64-NixOS:

```
11 packages built:
elmerfem inkcut inkcut.dist inkscape-extensions.inkcut mirtk python310Packages.enamlx python310Packages.enamlx.dist python310Packages.pythonocc-core python311Packages.enamlx python311Packages.enamlx.dist python311Packages.pythonocc-core

```

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 255928` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>openmodelica.combined</li>
    <li>openmodelica.omedit</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>cloudcompare</li>
    <li>elmerfem</li>
    <li>entwine</li>
    <li>grass</li>
    <li>inkcut</li>
    <li>inkcut.dist</li>
    <li>inkscape-extensions.inkcut</li>
    <li>mirtk</li>
    <li>openscenegraph</li>
    <li>pdal</li>
    <li>python310Packages.enamlx</li>
    <li>python310Packages.enamlx.dist</li>
    <li>python310Packages.pythonocc-core</li>
    <li>python311Packages.enamlx</li>
    <li>python311Packages.enamlx.dist</li>
    <li>python311Packages.pythonocc-core</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
    <li>simgear</li>
    <li>sumo</li>
  </ul>
</details>

**Pull request merged**